cmake/Modules/Finddpdk.cmake

```diff
@@ -0,0 +1,252 @@
+# Try to find dpdk
+#
+# Once done, this will define
+#
+# dpdk::dpdk
+# dpdk_FOUND
+# dpdk_INCLUDE_DIR
+# dpdk_LIBRARIES
+
+find_package(PkgConfig QUIET)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(dpdk QUIET libdpdk)
+endif()
+
+if(dpdk_INCLUDE_DIRS)
+  # good
+elseif(TARGET dpdk::dpdk)
+  get_target_property(dpdk_INCLUDE_DIRS dpdk::dpdk
+                      INTERFACE_INCLUDE_DIRECTORIES)
+else()
+  find_path(
+    dpdk_config_INCLUDE_DIR rte_config.h
+    PATH_SUFFIXES dpdk include)
+  find_path(
+    dpdk_common_INCLUDE_DIR rte_common.h
+    PATH_SUFFIXES dpdk include)
+  set(dpdk_INCLUDE_DIRS "${dpdk_config_INCLUDE_DIR}")
+  if(NOT dpdk_config_INCLUDE_DIR STREQUAL dpdk_common_INCLUDE_DIR)
+    list(APPEND dpdk_INCLUDE_DIRS "${dpdk_common_INCLUDE_DIR}")
+  endif()
+endif()
+
+# Order is important ! Taken from the DPKT pkg-config file
+set(components
```

Comment:
This is a bit tricky because different versions of DPDK may have a different set of libraries.
In the current build system, we only link with libraries that are required:
- For DPDK versions >= 20.11: https://github.com/seladb/PcapPlusPlus/blob/22269e8d40bebdbef225d63bf0a92a9bd94988d7/mk/PcapPlusPlus.mk.dpdk#L28
- For older DPDK versions: https://github.com/seladb/PcapPlusPlus/blob/22269e8d40bebdbef225d63bf0a92a9bd94988d7/mk/PcapPlusPlus.mk.dpdk_legacy#L4

I'm not sure how to do that with CMake, but there's probably a way to do so 🤔

Reply:
reworked this part